src/pnal.h

```diff
@@ -315,6 +335,24 @@ pnal_eth_handle_t * pnal_eth_init (
    pnal_eth_callback_t * callback,
    void * arg);
 
+/**
+ * Set the local port number for a network interface.
+ * @param handle           InOut: Ethernet handle
+ * @param loc_port_num     In:    Local port number.
+ *                                Valid range: 1 .. PNET_MAX_PORT.
+ * @return  0 if the operation succeeded.
+ *         -1 if an error occurred.
+ */
+int pnal_eth_set_local_port_num (pnal_eth_handle_t * handle, int loc_port_num);
```

Comment:
Does pnal really need to know the port number?

Reply:
Maybe not. Do you have some other suggestion how we can implement it?

Reply:
I haven't looked closely at the code so I can't say.

As I understand it, the (physical) port numbers are an abstraction that belong to the Profinet level. Exposing them to the low-level Ethernet layer seems wrong.

Reply:
I  wanted  a way to tag a network interface with a local port number. To make it more general the term local_port_num could be replaced with "tag". Since this is an abstraction layer for the p-net stack I think adding the get/set for local_port_num is ok.

I will find another solution.

Reply:
I think the change to be able to set an integer tag on each interface is good.